cmake/OpenCVGenPkgconfig.cmake

```diff
@@ -8,71 +8,89 @@
 #
 # ${BIN_DIR}/unix-install/opencv.pc -> For use *with* "make install"
 # -------------------------------------------------------------------------------------------
-set(prefix      "${CMAKE_INSTALL_PREFIX}")
-set(exec_prefix "\${prefix}")
-set(libdir      "") #TODO: need link paths for OpenCV_EXTRA_COMPONENTS
-set(includedir  "\${prefix}/${OPENCV_INCLUDE_INSTALL_PATH}")
-
-if(CMAKE_BUILD_TYPE MATCHES "Release")
-  set(ocv_optkind OPT)
-else()
-  set(ocv_optkind DBG)
-endif()
 
-#build the list of opencv libs and dependencies for all modules
-set(OpenCV_LIB_COMPONENTS "")
-set(OpenCV_EXTRA_COMPONENTS "")
-foreach(m ${OPENCV_MODULES_PUBLIC})
-  list(INSERT OpenCV_LIB_COMPONENTS 0 ${${m}_MODULE_DEPS_${ocv_optkind}} ${m})
-  if(${m}_EXTRA_DEPS_${ocv_optkind})
-    list(INSERT OpenCV_EXTRA_COMPONENTS 0 ${${m}_EXTRA_DEPS_${ocv_optkind}})
-  endif()
-endforeach()
+macro(fix_prefix lst)
+  set(_lst)
+  foreach(item ${${lst}})
+    if(TARGET ${item})
+      get_target_property(libloc ${item} LOCATION_${CMAKE_BUILD_TYPE})
+      get_filename_component(libname ${libloc} NAME_WE)
+      # hack for 3rdparty libraries named liblib<something>
+      if (libname MATCHES "^liblib")
+        string(REGEX REPLACE "^liblib" "lib" libname ${libname})
+      else()
+        string(REGEX REPLACE "^lib" "" libname ${libname})
+      endif()
```

Comment:
cmake removes all occurencies: for pattern "^lib" it will transform "liblibwebp" to "webp", but actual library is named "liblibwebp" (when compiled from OpenCV source tree).